ssl/t1_lib.c

```diff
@@ -909,6 +909,29 @@ static int tls1_in_list(uint16_t id, const uint16_t *list, size_t listlen)
     return 0;
 }
 
+int tls1_get0_implemented_groups(int min_proto_version, int max_proto_version,
+                                 TLS_GROUP_INFO *grps, size_t num, long all,
+                                 STACK_OF(OPENSSL_CSTRING) *out)
+{
+    uint16_t id = 0;
+
+    if (grps == NULL || out == NULL)
+        return 0;
+    for (; num-- > 0; ++grps) {
+        /* Skip aliases */
+        if (!all && id == grps->group_id)
```

Comment:
Not sure it is guaranteed that aliases will always be consecutive. They are in our providers...but what about third party providers?

Reply:
I've made the dedup somewhat order independent, though when multiple groups share the same id, I'm reporting just the first one on the original list, hoping that's the more canonical name. Otherwise, there's no way to know...